File: validate-binary-search-tree/validate-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        
        inf=float('inf')
        
        def helper(root,minv,maxv):
            
            if not root:
                return True
            
            if root.val>minv and root.val<maxv:
                return helper(root.left,minv,root.val) and helper(root.right,root.val,maxv)
            
        return helper(root,-inf,inf)
            
        

File: validate-binary-search-tree/validate-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isValidBST = function(root) {
    
    // inf=1/0
    
    
    
    var helper = function(minv,root,maxv){
        
        if(root===null){
            return true;
        }

        if(root.val>minv && root.val<maxv){
            return (helper(minv,root.left,root.val) && helper(root.val,root.right,maxv));
        }
        else{
            return false;
        }
    };
    
    
    return helper(Number.MIN_SAFE_INTEGER,root,Number.MAX_SAFE_INTEGER);
};
